### PR TITLE
Chore: remove testing on MySQL 5.7

### LINKED ISSUE #5364: Move to MySQL8
For some of our advanced queries, it would b really helpful to move entirely to MySQL 8.  Specifically, it would be nice to have [window functions](https://dev.mysql.com/doc/refman/8.0/en/window-functions.html) and partitioning.

We have a few places where we hack this together for the "simpler" queries for running balance.  However, if we want to calculate multiple running balances (as we do in the CMM calculations), we'll want to keep track of multiple running balances at a time.  I can't figure out an easy way to do this in MySQL 5.7, but it is trivial in MySQL 8:
```sql
-- pseudocode
SELECT 
  quantity,
  sum(quantity) over (partition by inventory_uuid order by day) as cumulative_sum
FROM stock_movement_status
```

These benefits (plus the utf8mb4 by default, etc) should make us take a serious look at moving entirely to MySQL 8.

~~We do have issues in our trial balance tests for MySQL 8 that should give us pause ... but in theory it's a misunderstanding on our part and we can figure them out.~~ UPDATE: solved this.

### PR DESCRIPTION
**Pull request by jniles**
This commit removes testing on MySQL 5.7 and also removes the documentation tied to MySQL 5.7.  It completes the transition to MySQL 8.

Closes #5364. 

-----
[View rendered docs/en/for-developers/installing-bhima.md](https://github.com/jniles/bhima/blob/chore-remove-testing-on-mysql-5.7/docs/en/for-developers/installing-bhima.md)
[View rendered docs/fr/for-developers/installing-bhima.md](https://github.com/jniles/bhima/blob/chore-remove-testing-on-mysql-5.7/docs/fr/for-developers/installing-bhima.md)

### CODE REVIEW

File: docs/en/for-developers/installing-bhima.md [hunk 1]
```diff
@@ -149,9 +149,7 @@ docker run --name mysql8 -p 3306:3306 \
   -e MYSQL_ROOT_PASSWORD=MyPassword \
   -d mysql:8 \
   --sql-mode='STRICT_ALL_TABLES,NO_UNSIGNED_SUBTRACTION' \
-  --default-authentication-plugin=mysql_native_password \
-  --character-set-server=utf8mb4 \
-  --collation-server=utf8mb4_unicode_ci
+  --default-authentication-plugin=mysql_native_password
```
Comment (jmcameron) [inline docs/en/for-developers/installing-bhima.md:152]: Do we still need the trouble-shooting section at the end of this document?

**Review by jmcameron — APPROVED** [review]
The script changes and documentation changes look goo.d
Was not able to run it to really test it.

**Comment by jniles** [comment]
@jmcameron that was a good catch.  I've removed the troubleshooting section.  It only applied to MySQL anyway.

**Pull request merged**